src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import * as todoService from '../../api/todos';
+import { ErrorMessages } from '../../types/Errors';
+
+type FilterTypes = 'All' | 'Active' | 'Completed';
+
+interface Props {
+  todos: Todo[];
+  filterType: FilterTypes;
+  onFilterType: (v: FilterTypes) => void;
+  onTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  onLoading: React.Dispatch<React.SetStateAction<number[]>>;
+  onErrorMessage: (message: ErrorMessages) => void;
+}
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filterType,
+  onFilterType,
+  onTodos,
+  onLoading,
+  onErrorMessage,
+}) => {
+  const handleFilterAll = () => {
+    onFilterType('All');
+  };
+
+  const handleFilterActive = () => {
+    onFilterType('Active');
+  };
+
+  const handleFilterCompleted = () => {
+    onFilterType('Completed');
+  };
+
+  function deleteCompletedTodos() {
+    todos.map(todo => {
+      if (todo.completed) {
+        onLoading(prev => [...prev, todo.id]);
+        todoService
+          .deleteTodos(todo.id)
+          .then(() =>
+            onTodos(currentTodos =>
+              currentTodos.filter(currentTodo => currentTodo.id !== todo.id),
+            ),
+          )
+          .catch(() => onErrorMessage(ErrorMessages.Delete))
+          .finally(() =>
+            onLoading(prev => prev.filter(item => item !== todo.id)),
+          );
+      }
+    });
+  }
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {[...todos].filter(todo => !todo.completed).length} items left
```

Comment:
Move this logic to the variable and use it here
```suggestion
        {todos.filter(todo => !todo.completed).length} items left
```